xray/plot/utils.py

```diff
@@ -123,28 +123,47 @@ def _determine_cmap_params(plot_data, vmin=None, vmax=None, cmap=None,
 
     calc_data = np.ravel(plot_data[~pd.isnull(plot_data)])
 
+    # Setting center=False prevents a divergent cmap
+    divergent = center is not False
+
+    # Set center to 0 so math below makes sense but remember its state
+    center_is_none = False
+    if center is None:
+        center = 0
+        center_is_none = True
+
+    # Setting non-centered vmin, vmax prevents a divergent cmap
```

Comment:
I would suggest doing this a little differently: if both vmin and vmax are not None, always set `divergent = False`. Relying on the values being close seems error prone, and if I want a divergent colormap, I can simply provide only one of `vmin` and `vmax`.